xbmc/utils/SortUtils.cpp

```diff
@@ -965,6 +965,11 @@ void SortUtils::Sort(SortBy sortBy, SortOrder sortOrder, SortAttribute attribute
     items.erase(items.begin(), items.begin() + limitStart);
     limitEnd -= limitStart;
   }
+  else if (limitStart > 0)
```

Comment:
I would probably group the conditions a bit differently to make it more obvious that the existing and the new case concern `limitStart > 0` by using
```cpp
if (limitStart > 0)
{
  if (static_cast<size_t>(limitStart) < items.size())
  {
    items.erase(items.begin(), items.begin() + limitStart);
    limitEnd -= limitStart;
  }
  else
  {
    items.erase(items.begin(), items.end());
    limitEnd = 0;
  }
}
```